rbac/default-role-manager/role_manager.go

```diff
@@ -40,40 +41,57 @@ func NewRoleManager(maxHierarchyLevel int) rbac.RoleManager {
 	rm.allRoles = &sync.Map{}
 	rm.maxHierarchyLevel = maxHierarchyLevel
 	rm.hasPattern = false
-
+	rm.hasDomainPattern = false
 	return &rm
 }
 
 // e.BuildRoleLinks must be called after AddMatchingFunc().
 //
 // example: e.GetRoleManager().(*defaultrolemanager.RoleManager).AddMatchingFunc('matcher', util.KeyMatch)
-func (rm *RoleManager) AddMatchingFunc(name string, fn MatchingFunc) {
-	rm.hasPattern = true
-	rm.matchingFunc = fn
+// If need to add Matchingfunc to the domain, add true at the end.
+// example: e.GetRoleManager().(*defaultrolemanager.RoleManager).AddMatchingFunc('matcher', util.KeyMatch, true)
+func (rm *RoleManager) AddMatchingFunc(name string, fn MatchingFunc, forDomain ...bool) error {
```

Comment:
I don't want to add new parameters, but I'm not sure can we support all the args of g at the same time.

Reply:
It is not yet to support the 2nd parameter, but the 1st and 3rd can be used at the same time, through the new commit

Reply:
What do you think this API should be, domain and sub/obj may be different matching function, we must add them apart

Reply:
Sorry @dovics , I made the wrong decision. You are right, we must use different matching function for each args.